### PR TITLE
Datns/guide

### PR DESCRIPTION
**Pull request by datns**
page How to play

### CODE REVIEW

File: launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Image, StyleSheet, View } from 'react-native';
+import { DimensionState, dimensionState, Text } from '@metacraft/ui';
+import { battlefield } from 'screens/Guide/content';
+import ConceptButton from 'screens/Guide/Dashboard/ConceptButton';
+import { sharedStyle } from 'screens/Guide/shared';
+import { useSnapshot } from 'utils/hook';
+import resources from 'utils/resources';
+
+const styles = StyleSheet.create({
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx:10]: Please move styles to the end of component. As react native recommend, "moving styles away from the render function, you're making the code easier to understand"

File: launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	conceptContainer: {
+		flexDirection: 'row',
+		flexWrap: 'wrap',
+		justifyContent: 'space-evenly',
+		marginTop: 40,
+	},
+});
+
+const BattlefieldOverview: React.FC<Record<string, unknown>> = () => {
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx:19]: Could use `FC` (when there is no Props) or `FC<Props>` (when there is Props)

File: launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	const [selectedConcept, setSelectedConcept] = React.useState<number>(0);
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx:20]: IMO, `import { FC, useState }` is less verbose, improving readability, less typing and clean code

File: launcher/screens/Guide/Dashboard/ConceptButton.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+const ConceptButton: FC<Props> = ({
+	label,
+	icon,
+	isFirst,
+	isLast,
+	onPress,
+	isSelected,
+}) => {
+	const containerStyle = StyleSheet.flatten([
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/ConceptButton.tsx:47]: @fluctlight-kayaba, is flatten necessary here? Could I use:
`const containerStyle = {
   ...styles.container,
  marginLeft: isFirst ? 0 : 20,
  marginRight: isLast ? 0 : 20,
}`

File: launcher/screens/Guide/Dashboard/Footer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		height: (width * 385) / 1555,
+		alignItems: 'center',
+		justifyContent: 'center',
+	} as ViewStyle;
+
+	const imageSize = {
+		width: '50%',
+		aspectRatio: 884 / 222,
+	} as ImageStyle;
+	return (
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/Footer/index.tsx:30]: Should separate function return with the above by 1 empty line

File: launcher/screens/Guide/Dashboard/Footer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		<ImageBackground
+			source={resources.guide.footerMainBackground}
+			style={[iStyles.wideContainer, backgroundContainer]}
+		>
+			<View style={imageSize}>
+				<ImageBackground
+					source={resources.guide.footerContentBackground}
+					style={{ flex: 1, alignItems: 'center', paddingTop: 20 }}
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/Footer/index.tsx:38]: Avoid using incline style if there is more than 2 fields, use `StyleSheet`  instead

File: launcher/screens/Guide/shared.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+		...iStyles.wideContainer,
+	},
+	heading: {
+		fontFamily: 'Volkhov',
+		fontWeight: '600',
+		color: '#fff',
+		textAlign: 'center',
+	},
+	subHeading: {
+		fontFamily: 'Poppins',
```
Comment (FME849) [inline launcher/screens/Guide/shared.ts:26]: The default font family of `<Text>` component is already `Poppins`, and default color in theme has been changed to `#fff`

File: launcher/screens/Guide/shared.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+		maxWidth: 1100,
+		paddingVertical: 40,
+		color: 'white',
+	},
+	buttonText: {
+		textAlign: 'center',
+		color: '#fff',
+	},
+	textShadow: {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (FME849) [inline launcher/screens/Guide/shared.ts:36]: @fluctlight-kayaba, should eslint disable is used?

**Review by FME849 — COMMENTED** [review]
@fluctlight-kayaba please review again, thank you.

File: launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (fluctlight-kayaba) [reply launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx:19]: yep, then in this case.. it should be:
```typescript
const BattlefieldOverview: FC = () => {
  // implementation
}
```

File: launcher/screens/Guide/Dashboard/Cards/index.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+							icon={concept.icon}
+							isFirst={index === 0}
+							isLast={index === concepts.length - 1}
+							key={concept.label}
+							onPress={() => setSelectedConcept(index)}
+						/>
+					);
+				})}
+			</View>
+			<Text style={styles.cardInfo}>{concepts[selectedConcept].content}</Text>
```
[the remark this answers is not in the document]
Comment (fluctlight-kayaba) [reply launcher/screens/Guide/Dashboard/Cards/index.tsx:151]:
```typescript
import { Markdown } from '@metacraft/ui';

const MyComp () => {
  const markdownContent = `
    formatted text content under Markdown is much easier, like **bold**, *italic*, or

    - a list
    - is possible
    - as well
    - and more to explore, see Source code: [here](https://github.com/cocrafts/metacraft-ui/blob/master/src/components/Markdown/index.tsx).
  `;

  return <Markdown content={markdownContent} />;
}
```

File: launcher/screens/Guide/shared.ts (hunk 8 above)
Comment (fluctlight-kayaba) [reply launcher/screens/Guide/shared.ts:36]: should not use it in most of the case -- but that's fine if we use it when we must

File: launcher/screens/Guide/Dashboard/Cards/index.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (datns) [reply launcher/screens/Guide/Dashboard/Cards/index.tsx]: image
I facing a warning about type so I add this comment to ignore it

File: launcher/screens/Guide/Dashboard/Cards/index.tsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+				<Text>{des}</Text>
+				{additional?.map((item) => {
+					return (
+						<React.Fragment key={item.title}>
+							<Text style={styles.propertyTitle}>{item.title}</Text>
+							<Text style={styles.propertyContent}>{item.text}</Text>
+						</React.Fragment>
+					);
+				})}
+			</>
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/Cards/index.tsx:34]: Avoid using empty tag, use `<Fragment></Fragment>` instead

File: launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+		);
+	};
+
+	const renderImage = (source: number) => {
+		return (
+			<View style={styles.imageWrapper}>
+				<Image source={source} style={imageSize} resizeMode="contain" />
+			</View>
+		);
+	};
```
Comment (FME849) [inline launcher/screens/Guide/Dashboard/BattlefieldOverview/index.tsx:30]: Image source should be `ImageSource` type instead of `number`

**Review by FME849 — APPROVED** [review]

**Review by fluctlight-kayaba — APPROVED** [review]

**Pull request merged**